### PR TITLE
Nick/adreview

### PR DESCRIPTION
**Pull request by grittygrease**

### CODE REVIEW

File: draft-ietf-tls-exported-authenticator.md [hunk 1]
```diff
@@ -156,28 +167,34 @@ sender:
...
+information at this stage since the application-supplied context
+is included in the authenticator itself.  The length of the exported value is equal to the length of the output of
+the hash function selected in TLS for the pseudorandom function (PRF).  Exported authenticators cannot be
+used with cipher suites that do not use the TLS PRF and have not defined
+a hash function for this purpose.  This hash is referred to as the authenticator hash.
+
+Let's rephrase this as "Exported Authenticators MUST NOT be generated or
+accepted on connectons not using the extended master secret {{!RFC7627}}
+extension, to avoid key synchronization attacks", to be very explicit
+about the prohibited behavior.
```
Comment (MikeBishop) [inline draft-ietf-tls-exported-authenticator.md:182]: This paragraph appears to be feedback containing a quoted suggestion, not the resulting text.

File: draft-ietf-tls-exported-authenticator.md [hunk 2]
```diff
@@ -202,15 +219,17 @@ In addition to "signature_algorithms" and "signature_algorithms_cert",
 the "server_name" {{!RFC6066}}, "certificate_authorities"
 (Section 4.2.4. of {{!TLS13}}), and "oid_filters"
 (Section 4.2.5. of {{!TLS13}}) extensions are used to guide certificate
-selection.  The extensions, or others that might affect certificate selection,
-are taken from the authenticator request if present, or the TLS handshake if not.
+selection.
 
-Alternative certificate formats such as {{!RFC7250}} Raw Public Keys
-are not supported in this version of the specification.
+Only the X509 certificate type defined in {{!TLS13}} is supported.
```
Comment (MikeBishop) [inline draft-ietf-tls-exported-authenticator.md:228]: X.509, no?

**Review by MikeBishop — APPROVED** [review]
Several really long lines might benefit from rewrapping, but only small nits on the changes here.

File: draft-ietf-tls-exported-authenticator.md (hunk 1 above)
Comment (martinthomson) [reply draft-ietf-tls-exported-authenticator.md:182]: Removing the framing might do, but watch for the typo on "connections"

File: draft-ietf-tls-exported-authenticator.md [hunk 3]
```diff
@@ -35,44 +35,52 @@ informative:
...
-This document intends to replace much of the functionality of renegotiation
-in previous versions of TLS.  It has the advantages over renegotiation of not
-requiring additional on-the-wire changes during a connection.  For simplicity,
-only TLS 1.2 and later are supported.
+Versions of TLS prior to TLS 1.3 used renegotiation as a way to enable
+post-handshake client authentication given an existing TLS connection.
+The mechanism described in this document may be used to replace the
+post-handshake authentication functionality provided by renegotiation.
+Unlike renegotiation, exported Authenticator-based post-handshake
+authentication does not require any on-the-wire changes.
```
Comment (kaduk) [inline draft-ietf-tls-exported-authenticator.md]: We probably want to qualify the lack of changes to "at the TLS level" or "to the cryptographic keys used" or something like that, as there is clearly some change in the application's bits on the wire to use EAs.

File: draft-ietf-tls-exported-authenticator.md (hunk 1 above)
Comment (kaduk) [reply draft-ietf-tls-exported-authenticator.md:182]: Indeed.

File: draft-ietf-tls-exported-authenticator.md (hunk 2 above)
Comment (kaduk) [reply draft-ietf-tls-exported-authenticator.md:228]: Per https://tools.ietf.org/html/rfc8446#section-4.4.2 the description language's element is spelled this way.

File: draft-ietf-tls-exported-authenticator.md [hunk 4]
```diff
@@ -231,9 +250,10 @@ excludes all RSASSA-PKCS1-v1_5 algorithms and combinations of ECDSA and hash
 algorithms that are not supported in TLS 1.3.
 
 If an authenticator request is present, the signature algorithm MUST be chosen
-from one of the signature schemes in the authenticator request.  Otherwise, the
-signature algorithm used should be chosen from the "signature_algorithms" sent
-by the peer in the TLS handshake.
+from one of the signature schemes present in the authenticator request.     Otherwise, the signature algorithm used should be chosen
+from the "signature_algorithms" sent by the peer in the ClientHello of the TLS
+handshake. There are no available signature algorithms, then no
```
Comment (kaduk) [inline draft-ietf-tls-exported-authenticator.md]: Missing "If"
